### PR TITLE
samples: Bluetooth: Audio: Find device regardless of name type

### PR DESCRIPTION
**Pull request by alexsven**
- Fixes bug where the sample would only look at the first name type and skip the rest.

### CODE REVIEW

File: samples/bluetooth/bap_broadcast_sink/src/main.c [hunk 1]
```diff
@@ -1218,11 +1224,11 @@ static void broadcast_scan_recv(const struct bt_le_scan_recv_info *info, struct
 		 */
 		if (req_recv_state == NULL && strlen(CONFIG_TARGET_BROADCAST_NAME) > 0U) {
 			struct net_buf_simple buf_copy;
-			char name[NAME_LEN] = {0};
 
 			net_buf_simple_clone(ad, &buf_copy);
 			bt_data_parse(&buf_copy, data_cb, name);
-			if (!(is_substring(CONFIG_TARGET_BROADCAST_NAME, name))) {
+
+			if (!device_found) {
```
Comment (Thalley) [inline samples/bluetooth/bap_broadcast_sink/src/main.c:1231]: I think you are missing a declaration of `device_found` and passing it on the `data_cb` by modifying `bt_data_parse(&buf_copy, data_cb, name);` to be `bt_data_parse(&buf_copy, data_cb, &device_found);`

File: samples/bluetooth/bap_broadcast_sink/src/main.c [hunk 2]
```diff
@@ -1194,14 +1194,20 @@ static bool is_substring(const char *substr, const char *str)
...
+	bool *device_found = user_data;
+	char name[NAME_LEN] = {0};
 
 	switch (data->type) {
 	case BT_DATA_NAME_SHORTENED:
 	case BT_DATA_NAME_COMPLETE:
 	case BT_DATA_BROADCAST_NAME:
 		memcpy(name, data->data, MIN(data->data_len, NAME_LEN - 1));
-		return false;
+		if (!(is_substring(CONFIG_TARGET_BROADCAST_NAME, name))) {
```
Comment (Thalley) [inline samples/bluetooth/bap_broadcast_sink/src/main.c:1205]:
```suggestion
		if (is_substring(CONFIG_TARGET_BROADCAST_NAME, name)) {
```

**Review by Thalley — CHANGES_REQUESTED** [review]

File: samples/bluetooth/bap_broadcast_sink/src/main.c (hunk 1 above)
Comment (alexsven) [reply samples/bluetooth/bap_broadcast_sink/src/main.c:1233]: Sorry, was a bit quick on the push-button. Fixed

File: samples/bluetooth/bap_broadcast_sink/src/main.c (hunk 2 above)
Comment (alexsven) [reply samples/bluetooth/bap_broadcast_sink/src/main.c]: Sorry, was a bit quick on the push-button. Fixed

**Review by Thalley — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**